### ISSUE TITLE
Frontend for ingesting new events into the Rust Community Events calandar

### ISSUE THREAD

**Issue by flaki** [rust-community/team#262]
As per our offline conversation with @badboy & @skade I have proposed a simple hosted form specially developed for this purpose by me. The form interface would be hosted on Glitch/Now and would be a reasonably straightforward method for people to submit new events into a calendar owned by the community team. This could be a Teamup subcalendar (as per #257), or even the [Google calendar](https://calendar.google.com/calendar/embed?showTitle=0&showPrint=0&showTabs=0&showCalendars=0&mode=AGENDA&height=400&wkst=1&bgcolor=%23FFFFFF&src=apd9vmbc22egenmtu5l6c5jbfc%40group.calendar.google.com&color=%23691426&ctz=Europe%2FMadrid) currently in use for this purpose.

The user interface would replicate the inputs of the [current Google form in use](https://docs.google.com/forms/d/e/1FAIpQLSf52YXGhqBaHtCXtVna4iHYMK7IQaTqUW6V-ztsZC8C2TBInQ/viewform), and would make it possible to moderate input before it is posted to our calendar instance. Submissions would be emailed to the submitter as well who would have a unique link to edit these submissions later (edits would require moderation, too, and would go live after accepted by a moderator). Moderators would receive a notification about the submitted event.

Other alternatives considered was a [Calagator instance](https://rust-community-calendar.herokuapp.com/) which would also require some maintenance, but more importantly feature development (it currently lacks timezone support), which could be sponsored by the Rust project but would push out our timelines considerably.

**Comment by Manishearth** [comment]
So, FWIW Calendar already has functionality like this: https://support.google.com/calendar/answer/44105?hl=en

With this, we can set up a calendar that auto-accepts invites, and share the invite link to meetup organizers. The community calendar is already set up this way (I tested it), however people add events to it directly rather than via the invite link.

This cannot be moderated, sadly (we can delete things after the fact though). If we wish for it to be moderated, we can instead:

 - create a new google account for the calendar
 - have team members have access
 - share that email address out, but _do not_ set the calendar to autoaccept invites
 - let team members accept invitations as they come in

**Comment by skade** [comment]
I would love if there were a less black & white way of moderation. Generally, I trust people that have added an event or two and grave problems will end up being caught by the TWIR team.

**Comment by Manishearth** [comment]
In that case can we rely on the privacy of the calendar email address, and just keep an eye out when people get added?

**Comment by skade** [comment]
Just to be clear: would that still mean we have to add people manually?

**Comment by Manishearth** [comment]
> Just to be clear: would that still mean we have to add people manually

No, we respond to requests for access to the calendar by saying "please add `blahblahblahblah22u3983290uf@group.calendar.google.com` as a guest to any events you want to add to the rust calendar"

People will need a google calendar account to create events, still, but you can have a google calendar account without using gmail.

**Comment by Manishearth** [comment]
The dedicated email address option is where we tell people "please add `rust-community-calendar@gmail.com` as a guest" and then team members accept invites.
